webapp/exercise_admin/static/exercise_admin/exercise-admin.js

```diff
@@ -45,8 +45,86 @@ $(document).ready(function() {
     $('.popup > div').click(function(event) {
         event.stopPropagation();
     });
+    $('input[type=text].exercise-tag-input').each(function () {
+        change_tag_width(this);
+        if (this.value === "") {
+            remove_tag(this);
+        }
+    });
 });
 
+function highlight_parent_li(input_elem) {
+    var input = $(input_elem);
+    var parent = input.parent();
+    parent.css({'background' : '#cec'});
+    input.focusout(function() {
+        if (input.val() === '') {
+            remove_tag(input[0]);
+        } else if (!parent.is(':hover')) {
+            parent.css({'background' : '#ddd'});
+        }
+    });
+    parent.hover(function() {
+        parent.css({'background' : '#cec'});
+    }, function() {
+        if (!input.is(':focus')) {
+            parent.css({'background' : '#ddd'});
+        }
+    });
+}
+
+function change_tag_width(input_elem) {
+    var CHAR_WIDTH = 9;
+    var MAX_FITTING_LENGTH = 4;
+    var input = $(input_elem);
+    var val_len = input.val().length;
+    
+    if (val_len > MAX_FITTING_LENGTH) {
+        input.width((input_elem.value.length * CHAR_WIDTH) + 'px');
+    } else {
+        input.width(input.css('min-width'));
+    }
+}
+
+function add_tag() {
+    var MAX_COUNT = 20;
```

Comment:
We don't need a maximum tag count (and if we did, it would be enforced in the db schema).